### PR TITLE
fixes #266

### LINKED ISSUE #266: [Enhancement] The "Privacy and Terms" pop-up should be dismissed when pressing Esc key
**[Affected versions]:**
- Firefox 59.0.3 and up

*[Affected Platforms]:*
- All Windows
- All Mac
- All Linux

**[Prerequisites]:**

- Have a Firefox profile with the latest version of the "FirefoxColor" add-on (0.0.35 custom build) installed from the "about:debugging" page.

*[Steps to reproduce]:*
1. Open the browser with the profile from prerequisites.
2. Click the "Firefox Color" toolbar button.
3. Go to footer section and click on "Privacy" or "Terms" button.
4. Press Esc key.

**[Expected result]:**
- The pop-up is dismissed. 

**[Actual result]:**
- The pop-up remains open.

### PR DESCRIPTION
**Pull request by ankiiitraj**
Added a function to register keydown.

### CODE REVIEW

File: src/web/lib/components/Modal/index.js [hunk 1]
```diff
@@ -9,6 +9,17 @@ export const Modal = ({ toggleModal, displayModal, children }) => {
...
+
+  const handleKeyPress = event => {
+    if (event.keyCode === 27) {
+      toggleModal({ display: false });
+      console.log(event.keyCode);
+      document.removeEventListener("keydown", handleKeyPress);
+    }
+  };
+
+  document.addEventListener("keydown", handleKeyPress);
```
Comment (Rob--W) [inline src/web/lib/components/Modal/index.js]: You're adding a listener, and only removing it when the Esc key is pressed.

If the Esc key is not pressed, then the event listener will continue to exist, and over time the document will accumulate unused event listeners. As a result, when you press Esc, then `toggleModal` will be called multiple times, which may result in the lag that you've described.

Use `componentDidMount` and `componentWillUnmount` to correctly clean up the listeners, as I described in my comments in https://github.com/mozilla/FirefoxColor/issues/266

Comment (ankiiitraj) [reply src/web/lib/components/Modal/index.js]: Thanks for the guidance :)

**Comment by ankiiitraj** [comment]
Hi @Rob--W, I have tested the commit and it works fine now. Can you assign yourself as the reviewer and assign me the bug. Thanks.

**Comment by Rob--W** [comment]
FYI: you didn't have to close the pull request. Github automatically updates the PR, and I would have been able to review and approve the changes.

**Pull request closed without merging**